File: plugins/OrchidPosts/src/main/java/com/eden/orchid/posts/permalink/pathTypes/MonthNamePathType.java
package com.eden.orchid.posts.permalink.pathTypes;

import com.eden.orchid.api.theme.pages.OrchidPage;
import com.eden.orchid.posts.pages.PostPage;
import com.eden.orchid.posts.permalink.PermalinkPathType;

import javax.inject.Inject;
import java.time.Month;

public class MonthNamePathType extends PermalinkPathType {

    @Inject
    public MonthNamePathType() {
        super(100);
    }

    @Override
    public boolean acceptsKey(OrchidPage page, String key) {
        return key.equals("monthName");
    }

    @Override
    public String format(OrchidPage page, String key) {
        if(page instanceof PostPage) {
            return Month.of(((PostPage) page).getMonth()).toString();
        }

        return null;
    }

}

File: plugins/OrchidPosts/src/main/kotlin/com/eden/orchid/posts/permalink/pathtypes/MonthNamePathType.kt
package com.eden.orchid.posts.permalink.pathtypes

import com.eden.orchid.api.theme.pages.OrchidPage
import com.eden.orchid.api.theme.permalinks.PermalinkPathType
import com.eden.orchid.posts.pages.PostPage
import java.time.Month
import javax.inject.Inject

class MonthNamePathType
@Inject
constructor() : PermalinkPathType(100) {

    override fun acceptsKey(page: OrchidPage, key: String): Boolean {
        return key == "monthName"
    }

    override fun format(page: OrchidPage, key: String): String? {
        return if (page is PostPage) {
            Month.of(page.month).toString()
        } else page.publishDate.month.toString()

    }

}

